builtin/providers/kubernetes/patch_operations.go

```diff
@@ -0,0 +1,135 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"reflect"
+	"sort"
+	"strings"
+)
+
```

Comment:
So the struct type is called `PatchOperation` and the word _patch_ comes from PATCH HTTP method, or more specifically from [JSON Patch](https://tools.ietf.org/html/rfc6902) - so I think it's the right name.

I think eventually we could use these helpers cross providers, but it depends on the exact implementation of the SDK.